graphql/chat/mutations.ts

```diff
@@ -1,20 +1,43 @@
-import { Arg, Ctx, Mutation, Resolver } from 'type-graphql';
-import { Lecture as Appointment } from '../generated';
+import { Arg, Authorized, Ctx, Mutation, Resolver } from 'type-graphql';
 import { Role } from '../../common/user/roles';
+import * as GraphQLModel from '../generated/models';
 import { GraphQLContext } from '../context';
 import { AuthorizedDeferred, hasAccess } from '../authorizations';
 import { getLogger } from '../../common/logger/logger';
 import { prisma } from '../../common/prisma';
+import { getOrCreateChatUser, getOrCreateConversation } from '../../common/chat';
+import { getUser, getUserTypeAndIdForUserId } from '../../common/user';
+import { getMatchByMatchees, getUserIdsForChatParticipants, getUsersForChatParticipants } from '../../common/chat/helper';
 
 const logger = getLogger('MutateChatResolver');
-
-@Resolver(() => Appointment)
+@Resolver((of) => GraphQLModel.Match)
```

Comment:
Hm, can we omit the type here? I don't think a resolver type makes sense (as it is 'Chat' conceptually which is not an entity)